src/main/java/org/openrewrite/java/jackson/codehaus/ReplaceSerializationConfigAnnotationIntrospector.java

```diff
@@ -0,0 +1,79 @@
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.template.internal.AbstractRefasterJavaVisitor;
+import org.openrewrite.java.tree.J;
+
+import static org.openrewrite.java.template.internal.AbstractRefasterJavaVisitor.EmbeddingOption.SHORTEN_NAMES;
+
+public class ReplaceSerializationConfigAnnotationIntrospector extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "Migrate serialization annotation processor";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Migrate serialization annotation processor to use the codehaus config method.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        JavaVisitor<ExecutionContext> javaVisitor = new AbstractRefasterJavaVisitor() {
+            final JavaTemplate before = JavaTemplate
+                    .builder("#{mapper:any(org.codehaus.jackson.map.ObjectMapper)}.getSerializationConfig().setAnnotationIntrospector(#{introspector:any(org.codehaus.jackson.map.AnnotationIntrospector)});")
+                    .javaParser(JavaParser.fromJavaVersion().classpath(JavaParser.runtimeClasspath()))
+                    .build();
+            final JavaTemplate after = JavaTemplate
+                    .builder("#{mapper:any(com.fasterxml.jackson.databind.ObjectMapper)}.setConfig(#{mapper}.getSerializationConfig().with(#{introspector:any(com.fasterxml.jackson.databind.AnnotationIntrospector)}));")
+                    .javaParser(JavaParser.fromJavaVersion().classpath(JavaParser.runtimeClasspath()))
+                    .build();
+
+            @Override
+            public J visitMethodInvocation(J.MethodInvocation elem, ExecutionContext ctx) {
+                JavaTemplate.Matcher matcher = before.matcher(getCursor());
+                if (matcher.find()) {
+                    return embed(
+                            after.apply(getCursor(), elem.getCoordinates().replace(), matcher.parameter(0), matcher.parameter(1)),
+                            getCursor(),
+                            ctx,
+                            SHORTEN_NAMES
+                    );
+                }
+                return super.visitMethodInvocation(elem, ctx);
+            }
+        };
+        return Preconditions.check(
+                Preconditions.and(
+                        new UsesType<>("org.codehaus.jackson.map.ObjectMapper", true),
+                        new UsesMethod<>("org.codehaus.jackson.map.MapperConfig setAnnotationIntrospector(..)", true),
+                        new UsesMethod<>("org.codehaus.jackson.map.ObjectMapper getSerializationConfig(..)", true)
```

Comment:
The uses type on ObjectMapper is implied when you also have the UsesMethod; no need to have & evaluate both then.
```suggestion
                        new UsesMethod<>("org.codehaus.jackson.map.MapperConfig setAnnotationIntrospector(..)", true),
                        new UsesMethod<>("org.codehaus.jackson.map.ObjectMapper getSerializationConfig(..)", true)
```